in-progress/8757-fees/design.md

```diff
@@ -0,0 +1,265 @@
+# Testnet Fees
+
+|                      |                                                                    |
+| -------------------- | ------------------------------------------------------------------ |
+| Issue                | https://github.com/AztecProtocol/aztec-packages/issues/8757        |
+| Owners               | @just-mitch @LHerskind                                             |
+| Approvers            | @Maddiaa0 @PhilWindle @dbanks12 @nventuro @joeandrews @aminsammara |
+| Target Approval Date | 2024-10-24                                                         |
+
+## Mana and Costs
+
+Refer to the unit of work on L2 as "mana", analogous to gas on L1. This has been referred to as "L2 gas" in the past.
+This is also separate from the notion of "fee juice", which is conceptually a token with a balance, analogous to ETH.
+
+In this document, we will use the term "fee asset" to refer to the unit of account for fees; that is, a transaction accumulates a cost in mana, which is converted to the fee asset according to the current price of mana.
+
+A transaction incurs the following costs:
+1. Proposers/validators simulate the transaction
+2. Proposers publish pending blocks to L1
+3. Proposers/validators/nodes update their state
+4. Provers generate the avm proofs for the transaction
+5. Provers generate the rollup proofs for the blocks/epoch
+6. Provers verify the epoch proof on L1
+
+Some costs are independent of the computation or data of the transaction itself, specifically numbers 2, 4, and 5; regardless, they must be covered by the transaction.
```

Comment:
The addition of the state updates (3) means that the below should also be updated to be 2, 5 and 6? 
```suggestion
Some costs are independent of the computation or data of the transaction itself, specifically numbers 2, 5, and 6; regardless, they must be covered by the transaction.
```